### PR TITLE
PAAS-1170: Use vault to fetch haproxy configurations

### PR DESCRIPTION
**Pull request by amitc-jahia**
JIRA issue: https://jira.jahia.org/browse/PAAS-1170

Short description:

### CODE REVIEW

File: utils/rewrite-rules.yml [hunk 1]
```diff
@@ -11,75 +11,55 @@ globals:
...
-  - cmd [bl]: |-
-        python3 /usr/local/bin/set-rewrite-rules.py "${globals.haproxy_frontend_file}" "${settings.rewrite-rules.toBase64()}"
-    user: root
-  - if ("${response.responses.errOut}" != ""):
-      - getRewritesRules
-      - return:
-          type: error
-          message: "An error occured when setting rewrite rules."
-          currentRules: "${globals.rewrites}"
+        if (! rpm --quiet -q jq);
```
Comment (Ronan-Gorain) [inline utils/rewrite-rules.yml:23]: As (now) mentioned in https://jira.jahia.org/browse/PAAS-1170 , the token and the vault hostname are not provided by the front. They already exists on bl as env var for the haproxy user, and in /.jelenv for root user.
The token can be fetched with something like this (didn't test)
`
source /.jelenv
token=$(curl  -XPOST https://$VAULT_CLUSTER_URL/v1/auth/approle/login --data '{"secret_id": "$VAULT_SECRET_ID", "role_id": "$VAULT_ROLE_ID"}' |jq -r .auth.client_token)`

File: utils/rewrite-rules.yml [hunk 2]
```diff
@@ -11,75 +11,55 @@ globals:
...
-          type: error
-          message: "${this.error-out}"
-          currentRules: "${globals.rewrites}"
+      user: root
+    - if ("${response.responses.errOut}" != ""):
+        - set:
+            error-out: "${response.responses.out} -- ${response.responses.errOut}"
+        - log: "reload failed"
+        - cmd [bl]: |-
+            mv "${globals.haproxy_backup_file}" "${globals.haproxy_frontend_file}"
```
Comment (Ronan-Gorain) [inline utils/rewrite-rules.yml:43]: We still need to return an error with the haproxy reload error output as it was done in the previous version

**Review by Ronan-Gorain — CHANGES_REQUESTED** [review]

**Comment by amitc-jahia** [comment]
Requested Changes have been made. Please review.

File: utils/rewrite-rules.yml [hunk 3]
```diff
@@ -11,75 +11,50 @@ globals:
...
-          currentRules: "${globals.rewrites}"
+        if (! rpm --quiet -q jq);
+        then
+          yum install jq -y
+        fi
+        source /.jelenv
+        VAULT_TOKEN=$(curl -s -XPOST $VAULT_CLUSTER_URL/v1/auth/approle/login --data '{"role_id": "'$VAULT_ROLE_ID'", "secret_id": "'$VAULT_SECRET_ID'"}' | jq -r .auth.client_token)
+        VAULT_SECRET_PATH="${settings.vault_secret_path}"
+        rewrite_rules=$(curl -s -H "X-Vault-Token: $VAULT_TOKEN" $VAULT_CLUSTER_URL/v1/kv/data/$VAULT_SECRET_PATH | jq -r .data.data[])
+        echo "$(awk -v rewriterules="$rewrite_rules" '/START_REWRITES/{print;print rewriterules;next}1' ${globals.haproxy_frontend_file})" > "${globals.haproxy_frontend_file}"
```
Comment (laurentfufu) [inline utils/rewrite-rules.yml:31]: they are a limits to a command line set at kernel level and that seems to be 2 MiB for _argv_ these days
so not sure it's a god id to use shell variable for potentially huge data

**Review by laurentfufu — CHANGES_REQUESTED** [review]

**Review by Ronan-Gorain — CHANGES_REQUESTED** [review]

File: utils/rewrite-rules.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (laurentfufu) [reply utils/rewrite-rules.yml:31]: didn't read the awk because of my previous comment but you right indeed

**Comment by amitc-jahia** [comment]
Following Changes made as requested :

1. Instead of using shell variable, the response from curl is dumped into a file and using the file to read the rules.
2. Using sed with expression script to remove all the content between the 2 markers and add the contents of rules file in between the markers.

Please review.

**Comment by Ronan-Gorain** [comment]
Seems good to me but I'm not good enough to understand your sed command :) I'll let Laurent validate

**Review by Ronan-Gorain — APPROVED** [review]

**Review by laurentfufu — APPROVED** [review]

**Pull request merged**